src/view/film-details.js

```diff
@@ -187,62 +202,147 @@ const createFilmDetailsTemplate = (film) => {
   );
 };
 
-export default class FilmDetails extends Abstract {
+export default class FilmDetails extends Smart {
   constructor(film) {
     super();
-    this._film = film;
-    this._toCloseClickHandler = this._toCloseClickHandler.bind(this);
+    this._state = FilmDetails.parseFilmToData(film);
+
     this._watchListClickHandler = this._watchListClickHandler.bind(this);
     this._watchedClickHandler = this._watchedClickHandler.bind(this);
     this._favoriteClickHandler = this._favoriteClickHandler.bind(this);
+    this._closeDetailsClickHandler = this._closeDetailsClickHandler.bind(this);
+    this._emotionClickHandler = this._emotionClickHandler.bind(this);
+    this._commentInputHandler = this._commentInputHandler.bind(this);
+    this.setInnerHandlers();
+
+    this._scrollPosition = 0;
   }
 
   getTemplate() {
-    return createFilmDetailsTemplate(this._film);
+    return createFilmDetailsTemplate(this._state);
   }
 
-  _toCloseClickHandler() {
-    this._callback.click();
+  _closeDetailsClickHandler() {
+    this._callback.toCloseClick();
   }
 
   _watchListClickHandler() {
-    this._callback.clickwatchList();
+    this._callback.watchListClick(this._state);
   }
 
   _watchedClickHandler() {
-    this._callback.clickWatched();
+    this._callback.watchedClick(this._state);
   }
 
   _favoriteClickHandler() {
-    this._callback.clickFavorite();
+    this._callback.favoriteClick(this._state);
+  }
+
+  _emotionClickHandler(evt) {
+    evt.preventDefault();
+    this._scrollPosition = this.getElement().scrollTop;
+
+    this.updateState({ emotion: evt.target.value });
+
+    this.getElement().scrollTop = this._scrollPosition;
+
+    const emojiItems = this.getElement().querySelectorAll('.film-details__emoji-item');
+    emojiItems.forEach((emotion) => {
+      if(emotion.value === evt.target.value){
+        emotion.setAttribute('checked', 'true');
+      }
+    });
```

Comment:
Можно как вариант чуть более декларативно записать
```suggestion
    [...emojiItems].find((emotion) => emotion.value === evt.target.value).setAttribute('checked', 'true');
```